lightning-invoice/src/utils.rs

```diff
@@ -427,15 +435,26 @@ fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Opt
 			htlc_minimum_msat: channel.inbound_htlc_minimum_msat,
 			htlc_maximum_msat: channel.inbound_htlc_maximum_msat,}])
 	};
-	// If all channels are private, return the route hint for the highest inbound capacity channel
-	// per counterparty node. If channels with an higher inbound capacity than the
-	// min_inbound_capacity exists, filter out the channels with a lower capacity than that.
+	// If all channels are private, prefer to return route hints which have a higher capacity than
+	// the payment value and where we're currently connected to the channel counterparty.
+	// Even if we cannot satisfy both goals, always ensure we include *some* hints, preferring
+	// those which meet at least one criteria.
 	filtered_channels.into_iter()
 		.filter(|(_counterparty_id, channel)| {
-			min_capacity_channel_exists && channel.inbound_capacity_msat >= min_inbound_capacity ||
-			!min_capacity_channel_exists
+			if online_min_capacity_channel_exists {
+				channel.inbound_capacity_msat >= min_inbound_capacity && channel.is_usable
+			} else if min_capacity_channel_exists && online_channel_exists {
```

Comment:
acc. to this comment "// If there are some online channels and some min_capacity channels, but no
// online-and-min_capacity channels, just include the min capacity ones and ignore
// online-ness."

check should only be on "min_capacity_channel_exists", online_channel_exists seems redundant